#
# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
#

Update HAL input device configuration for Solaris.   

Includes upstream commits:

commit 3575d9584edf35ec1720bc3755b6576a56613685
Author: Peter Hutterer <peter.hutterer@who-t.net>
Date:   Fri Jul 25 16:46:44 2008 +0930

    config: note that HAL options must be strings.
    
    Only strings are parsed by the server, all others are ignored. Doesn't 
    matter, specifying int options as strings works fine anyway.


commit d63ea510138c8b6de66184c78cda39ed9981fc1f
Author: Alan Coopersmith <alan.coopersmith@sun.com>
Date:   Wed Nov 5 14:52:29 2008 -0800

    Non-Linux OS'es should default to kbd driver, not now-dead keyboard driver


diff -urp -x '*~' -x '*.orig' config/x11-input.fdi config/x11-input.fdi
--- config/x11-input.fdi	2008-11-05 08:52:17.000000000 -0800
+++ config/x11-input.fdi	2009-01-14 19:06:14.942629000 -0800
@@ -26,6 +26,9 @@
 	    Do not specify "input.x11_options.Device" since "input.device"
 	    will be used automatically.
 
+            You MUST specify all options as strings, otherwise the server will
+            ignore them.
+
         Legacy Keys
 	     "input.xkb.rules"
              "input.xkb.model"
@@ -58,20 +61,34 @@
              string="Linux">
         <merge key="input.x11_driver" type="string">evdev</merge>
       </match>
+      <match key="/org/freedesktop/Hal/devices/computer:system.kernel.name"
+             string="SunOS">
+       <match key="input.device" contains="usb">
+        <merge key="input.x11_options.StreamsModule" type="string">usbms</merge>
+        <merge key="input.x11_options.Protocol" type="string">VUID</merge>
+       </match>
+      </match>
     </match>
 
     <match key="info.capabilities" contains="input.keys">
       <merge key="input.x11_options.XkbRules" type="string">base</merge>
 
       <!-- If we're using Linux, we use evdev by default (falling back to
-           keyboard otherwise). -->
-      <merge key="input.x11_driver" type="string">keyboard</merge>
+           kbd otherwise). -->
+      <merge key="input.x11_driver" type="string">kbd</merge>
       <merge key="input.x11_options.XkbModel" type="string">pc105</merge>
       <match key="/org/freedesktop/Hal/devices/computer:system.kernel.name"
              string="Linux">
         <merge key="input.x11_driver" type="string">evdev</merge>
         <merge key="input.x11_options.XkbModel" type="string">evdev</merge>
       </match>
+      <match key="/org/freedesktop/Hal/devices/computer:system.kernel.name"
+             string="SunOS">
+       <match key="input.device" contains="usb">
+        <merge key="input.x11_options.StreamsModule" type="string">usbkbm</merge>
+        <merge key="input.x11_options.Protocol" type="string">VUID</merge>
+       </match>
+      </match>
 
       <merge key="input.x11_options.XkbLayout" type="string">us</merge>
 
